### PR TITLE
feat: Build /20-04 page

### PR DESCRIPTION
**Pull request by petesfrench**
## Done

- Build /20-04 page based off of [copydoc](https://docs.google.com/document/d/1lzxAL9Wt24m5-p5qG9y190rOg3FY5zVdKLYLlBoBcLU/edit?tab=t.0) and [figma design](https://www.figma.com/design/ZAihKxF2wmj9wvoPSnPKmY/Commercial%3A-Ubuntu-Pro?node-id=5107-21524&node-type=canvas&t=e97GwrAiLwPAWxc0-0)
- Added some features to `_latest_news_strip.html`
- Bumped vanilla to `4.18.4`

## QA

- Check out [demo link](https://ubuntu-com-14559.demos.haus/20-04)
- Check the design matches the [figma design](https://www.figma.com/design/ZAihKxF2wmj9wvoPSnPKmY/Commercial%3A-Ubuntu-Pro?node-id=5107-21524&node-type=canvas&t=e97GwrAiLwPAWxc0-0)
- Check the content matches the [copydoc](https://docs.google.com/document/d/1BSlOqwe--4MoVvAEok6WssBfEfwiRSXN1dNWCoUffdY/edit?tab=t.0#heading=h.mgfeqagpepfl)
- Check there is correct application of vanilla, specifically focusing on macro usage
- Check on different screen sizes
- Check the pages where `_latest_news_strip.html` is implemented and ensure there is no difference

## Issue / Card

Fixes https://warthogs.atlassian.net/browse/WD-17287

### CODE REVIEW

**Comment by mattea-turic** [comment]
Thanks @petesfrench ! :) A few comments from me:

For the hero section section:
- [ ] Could you add a `<br>` after "support" pls?

For the "Expanded Security Maintenance (ESM) for 20.04 LTS" section:
- [ ] Could you replace the current full-width image under ESM with [this one](https://assets.ubuntu.com/v1/4fb3a7e7-esm-updated.png)? PM got back to me and is happy with the proposed change.

For the "What do you get with Ubuntu Pro for 20.04 LTS?" section:
- [ ] The CTA to the blog post can be removed (spoke to Sophie about this and it was a mistake in the file)
- [ ] Could you also add a HR above the first list item in the "Application packages" subsection?

For the "What customers say about ESM" section:
- [ ] There looks to be excess padding between the two testimonials – could you wrap the first testimonial in just a shallow pls?
- [ ] And for smaller screens, could you remove the shallow as I think this might help a little with hierarchy. 
- [ ] I'd also say (again, for smaller screens) remove the hr above the person's name & role, though iirc Madhur mentioned that this is something that isn't easily fixed as it's embedded within the comp

For the blog section:
- [ ] Could you apply deep padding there, since it's the last section of the page?
- [ ] For tablets, can the items span full width as they're right-aligned atm with empty columns

And a quick question about tables on smaller screens:
Screenshot 2024-12-11 at 17 31 33
Is there something we can do in place of truncation? No worries if not, just wanted to check with you

**Comment by mattea-turic** [comment]
Looks great, ty @petesfrench ! I'll +1, but just wanted to ask if you could add the line break after "support" in the h2 of the hero?

File: templates/20-04/index.html [hunk 1]
```diff
@@ -0,0 +1,287 @@
...
+
+  <section class="p-section">
+    <div class="row--50-50">
+      <hr class="p-rule" />
+      <div class="col">
+        <h2>Upgrading to Ubuntu 22.04 LTS, or Ubuntu 24.04 LTS fresh install</h2>
+      </div>
+      <div class="col">
+        <p>
+          Where it makes sense, you should transition workloads to later Ubuntu releases. You can either upgrade existing Ubuntu 20.04 LTS instances to Ubuntu 22.04 LTS or redeploy your workload onto a fresh Ubuntu 24.04 LTS instance. There is no direct upgrade path from 20.04 LTS to Ubuntu 24.04 LTS, so you can either move to Ubuntu 22.04 LTS and then to Ubuntu 24.04 LTS, or directly install Ubuntu 24.04 LTS.
```
Comment (britneywwc) [inline templates/20-04/index.html:115]: There should be a line break or another `p` here to separate these two paragraphs
```suggestion
          Where it makes sense, you should transition workloads to later Ubuntu releases. You can either upgrade existing Ubuntu 20.04 LTS instances to Ubuntu 22.04 LTS or redeploy your workload onto a fresh Ubuntu 24.04 LTS instance. 
<br />There is no direct upgrade path from 20.04 LTS to Ubuntu 24.04 LTS, so you can either move to Ubuntu 22.04 LTS and then to Ubuntu 24.04 LTS, or directly install Ubuntu 24.04 LTS.
```

File: templates/20-04/index.html [hunk 2]
```diff
@@ -0,0 +1,287 @@
...
+        </p>
+        <div class="p-cta-block">
+          <a href="https://ubuntu.com/server/docs/upgrade-introduction">Ubuntu Server upgrade guide&nbsp;&rsaquo;</a>
```
Comment (britneywwc) [inline templates/20-04/index.html:118]: The links are too close to each other here, maybe an inline list or something similar can be used here
Screenshot 2024-12-13 at 4 44 31 PM

File: templates/20-04/index.html [hunk 3]
```diff
@@ -0,0 +1,287 @@
...
+          <a href="https://ubuntu.com/tutorials/upgrading-ubuntu-desktop#1-before-you-start">Ubuntu Desktop upgrade guide&nbsp;&rsaquo;</a>
+        </div>
+      </div>
+    </div>
+  </section>
+
+  <section class="p-section">
+    <div class="row--50-50">
```
Comment (britneywwc) [inline templates/20-04/index.html:126]: There should be shallow padding wrapping the row to match Design
Screenshot 2024-12-13 at 4 47 08 PM

File: templates/20-04/index.html [hunk 4]
```diff
@@ -0,0 +1,287 @@
...
+      hi_def=True,
+      loading="lazy"
+      ) | safe
+      }}
+    </div>
+  </section>
+
+  {%- call(slot) vf_tiered_list(is_description_full_width_on_desktop=false, is_list_full_width_on_tablet=false) -%}
+    {%- if slot == 'title' -%}
+      <h2>What do you get with Ubuntu Pro for 20.04 LTS?</h2>
```
Comment (britneywwc) [inline templates/20-04/index.html:158]:
```suggestion
      <h2>What do you get <br class="u-hide-small u-hide-medium" /> with Ubuntu Pro for 20.04 LTS?</h2>
```

File: templates/20-04/index.html [hunk 5]
```diff
@@ -0,0 +1,287 @@
...
+    {%- endif -%}
+
+    {%- if slot == 'list_item_title_3' -%}
+
+      <h5>Application packages</h5>
+    {%- endif -%}
+
+    {%- if slot == 'list_item_description_3' -%}
+      <p>
+        Canonical provides application extended security maintenance to the binary packages that reside in Main and Universe. A few commonly-used packages include:
```
Comment (britneywwc) [inline templates/20-04/index.html:224]: There should also be line break or a separate paragraph here as per design
```suggestion
        Canonical provides application extended security maintenance to the binary packages that reside in Main and Universe. <br />A few commonly-used packages include:
```

File: templates/20-04/index.html [hunk 6]
```diff
@@ -0,0 +1,287 @@
...
+      <div class="p-section--shallow">
+        <p>
+          Standard Security Maintenance of an Ubuntu LTS release covers binary packages that reside in the Main Ubuntu repository for a period of five years. For continued security beyond the standard five-year maintenance period, Ubuntu Pro delivers security maintenance to a wide range of binary packages that are commonly used in cloud and server workloads on 64-bit x86 AMD/Intel architectures for a period of five years beyond the end of standard support.
+        </p>
+        <p>Additionally, Ubuntu Pro can also cover all packages in the Ubuntu Universe repository, until April 2030.</p>
+      </div>
+    {%- endif -%}
+
+    {%- if slot == 'list_item_title_1' -%}
+      <h5>Access to Canonical partner portal</h5>
```
Comment (britneywwc) [inline templates/20-04/index.html:171]:
```suggestion
      <h3 class="p-heading--5">Supported kernels</h3>
```

File: templates/20-04/index.html [hunk 7]
```diff
@@ -0,0 +1,287 @@
...
+    {%- endif -%}
+
+    {%- if slot == 'list_item_description_1' -%}
+      <p>
+        Resell the full range of Canonical's portfolio of private and public
+        cloud products: OpenStack, Kubernetes, IoT, support, security and
+        compliance for Ubuntu.
```
Comment (britneywwc) [inline templates/20-04/index.html:178]:
```suggestion
For continued Linux kernel security, Ubuntu Pro for Ubuntu 18.04 LTS includes support for the below versions:
```

File: templates/20-04/index.html [hunk 8]
```diff
@@ -0,0 +1,287 @@
...
+            </th>
+            <td>arm64, x86-64  (AMD/Intel) and s390x</td>
+            <td>5.15 (HWE)</td>
+          </tr>
+        </tbody>
+      </table>
+    {%- endif -%}
+
+    {%- if slot == 'list_item_title_2' -%}
+      <h5>Infrastructure packages</h5>
```
Comment (britneywwc) [inline templates/20-04/index.html:208]:
```suggestion
      <h3 class="p-heading--5">Infrastructure packages</h3>
```

File: templates/20-04/index.html [hunk 9]
```diff
@@ -0,0 +1,287 @@
...
+
+    {%- if slot == 'list_item_description_2' -%}
+      <p>
+        For Ubuntu 20.04 LTS, where technically feasible, Canonical provides extended security maintenance to all binary packages that reside in the Ubuntu Main Repository.
+      </p>
+    {%- endif -%}
+
+    {%- if slot == 'list_item_title_3' -%}
+
+      <h5>Application packages</h5>
```
Comment (britneywwc) [inline templates/20-04/index.html:219]:
```suggestion
      <h3 class="p-heading--5">Application packages</h3>
```

File: templates/20-04/index.html [hunk 10]
```diff
@@ -0,0 +1,287 @@
...
+      </ul>
+    {%- endif -%}
+  {%- endcall -%}
+
+  {% call(slot) vf_quote_wrapper(
+    title_text="What customers say about ESM",
+    quote_size="small",
+    quote_text="ESM literally saved our lives. It's allowing us to upgrade from 14.04 LTS at our own pace. It's taken the pressure off, and it also means we can tackle the Ubuntu upgrades at the same time as we roll out the new version of our platform.",
+    citation_source_name_text="Zivago Lee",
+    citation_source_title_text="Director of DevOps Engineeringr",
```
Comment (britneywwc) [inline templates/20-04/index.html:242]:
```suggestion
    citation_source_title_text="Director of DevOps Engineering",
```

File: templates/20-04/index.html [hunk 11]
```diff
@@ -0,0 +1,287 @@
...
+    citation_source_organisation_text="Interana",
+    is_shallow=True
+    ) -%}
+
+    {%- if slot == 'heading_link' -%}
+      <div class="p-section--shallow">
+        <a href="https://ubuntu.com/engage">All the success stories&nbsp;&rsaquo;</a>
```
Comment (britneywwc) [inline templates/20-04/index.html:249]: I can't find this link on the copy doc, is it missing?

File: templates/20-04/index.html [hunk 12]
```diff
@@ -0,0 +1,287 @@
...
+      <p>
+        <strong>Ubuntu 20.04 LTS (Focal Fossa)</strong> was released on April 23, 2020, with a strong focus on security and performance. Ubuntu 20.04 LTS also includes new security features like Kernel Self Protection measures, control flow integrity, and stack-clash protection, enhancing systemic enterprise security. Additional updates include support for WireGuard VPN, Secure Boot for protection against rootkits, FIDO for passwordless authentication, and CIS hardening on the server.
+      </p>
+    {%- endif -%}
+    {%- if slot == 'cta' -%}
+      <a href="/pro" class="p-button--positive">Get started with Ubuntu Pro for 20.04 LTS</a>
+    {%- endif -%}
+    {%- if slot == 'image' -%}
+      <div class="p-image-container--3-2 is-cover is-highlighted">
+        {{ image (
```
Comment (britneywwc) [inline templates/20-04/index.html:38]: Can you format the image tag pls?

File: templates/20-04/index.html (hunk 11 above)
Comment (mattea-turic) [reply templates/20-04/index.html:255]: @petesfrench Took from UX wireframe, I see now it wasn't in the copydoc, so I think you're good to just delete it. I'll just double-check with Sophie

File: templates/20-04/index.html [hunk 13]
```diff
@@ -0,0 +1,293 @@
...
+  <section class="p-section">
+    <div class="p-section--shallow">
+      <div class="row--50-50">
+        <hr class="p-rule" />
+        <div class="col">
+          <h2>Expanded Security Maintenance (ESM) for 20.04 LTS</h2>
+        </div>
+        <div class="col">
+          <p>
+            <a href="https://ubuntu.com/security/esm">Expanded Security Maintenance</a> (ESM) provides extended Linux kernel and open source security updates for the Ubuntu base OS, key infrastructure components, like Ceph, OpenStack and Kubernetes, as well as open source applications, like PostgreSQL and NGINX.
```
Comment (britneywwc) [inline templates/20-04/index.html:141]:
```suggestion
            <a href="/security/esm">Expanded Security Maintenance</a> (ESM) provides extended Linux kernel and open source security updates for the Ubuntu base OS, key infrastructure components, like Ceph, OpenStack and Kubernetes, as well as open source applications, like PostgreSQL and NGINX.
```

File: templates/20-04/index.html [hunk 14]
```diff
@@ -0,0 +1,293 @@
...
+          </p>
+          <p>
+            ESM is free on up to five machines and for broader enterprise use through an <a href="https://ubuntu.com/pro">Ubuntu Pro subscription</a>. The subscription also includes additional services such as FIPS-compliant modules and the Ubuntu Livepatch Service to apply critical kernel patches without unplanned downtime.
```
Comment (britneywwc) [inline templates/20-04/index.html:144]:
```suggestion
            ESM is free on up to five machines and for broader enterprise use through an <a href="/pro">Ubuntu Pro subscription</a>. The subscription also includes additional services such as FIPS-compliant modules and the Ubuntu Livepatch Service to apply critical kernel patches without unplanned downtime.
```

File: templates/20-04/index.html [hunk 15]
```diff
@@ -0,0 +1,293 @@
...
+          </p>
+          <div class="p-cta-block">
+            <a href="https://ubuntu.com/security/livepatch">Learn more about Ubuntu Livepatch&nbsp;&rsaquo;</a>
+          </div>
+        </div>
+      </div>
+    </div>
+    <div class="u-fixed-width">
+      {{ image (
```
Comment (britneywwc) [inline templates/20-04/index.html:153]: Can you format the image tag pls

File: templates/20-04/index.html [hunk 16]
```diff
@@ -0,0 +1,293 @@
...
+        </div>
+        <div class="col">
+          <p>
+            <a href="https://ubuntu.com/security/esm">Expanded Security Maintenance</a> (ESM) provides extended Linux kernel and open source security updates for the Ubuntu base OS, key infrastructure components, like Ceph, OpenStack and Kubernetes, as well as open source applications, like PostgreSQL and NGINX.
+          </p>
+          <p>
+            ESM is free on up to five machines and for broader enterprise use through an <a href="https://ubuntu.com/pro">Ubuntu Pro subscription</a>. The subscription also includes additional services such as FIPS-compliant modules and the Ubuntu Livepatch Service to apply critical kernel patches without unplanned downtime.
+          </p>
+          <div class="p-cta-block">
+            <a href="https://ubuntu.com/security/livepatch">Learn more about Ubuntu Livepatch&nbsp;&rsaquo;</a>
```
Comment (britneywwc) [inline templates/20-04/index.html:147]:
```suggestion
            <a href="/security/livepatch">Learn more about Ubuntu Livepatch&nbsp;&rsaquo;</a>
```

File: templates/20-04/index.html [hunk 17]
```diff
@@ -0,0 +1,291 @@
...
+  </section>
+
+  <section class="p-section">
+    <div class="row--50-50">
+      <hr class="p-rule" />
+      <div class="col">
+        <h2>Security updates for Ubuntu 20.04 LTS, with Ubuntu Pro</h2>
+      </div>
+      <div class="col">
+        {{ image (
```
Comment (britneywwc) [inline templates/20-04/index.html:89]: Can you format this image tag pls

File: templates/20-04/index.html [hunk 18]
```diff
@@ -0,0 +1,291 @@
...
+        <p>
+          Where it makes sense, you should transition workloads to later Ubuntu releases. You can either upgrade existing Ubuntu 20.04 LTS instances to Ubuntu 22.04 LTS or redeploy your workload onto a fresh Ubuntu 24.04 LTS instance.
+        </p>
+        <p>
+          There is no direct upgrade path from 20.04 LTS to Ubuntu 24.04 LTS, so you can either move to Ubuntu 22.04 LTS and then to Ubuntu 24.04 LTS, or directly install Ubuntu 24.04 LTS.
+        </p>
+        <div class="p-cta-block">
+          <ul class="p-inline-list">
+            <li class="p-inline-list__item">
+              <a href="https://ubuntu.com/server/docs/upgrade-introduction">Ubuntu Server upgrade guide&nbsp;&rsaquo;</a>
```
Comment (britneywwc) [inline templates/20-04/index.html:121]:
```suggestion
              <a href="/server/docs/upgrade-introduction">Ubuntu Server upgrade guide&nbsp;&rsaquo;</a>
```

File: templates/20-04/index.html [hunk 19]
```diff
@@ -0,0 +1,291 @@
...
+            </li>
+            <li class="p-inline-list__item">
+              <a href="https://ubuntu.com/tutorials/upgrading-ubuntu-desktop#1-before-you-start">Ubuntu Desktop upgrade guide&nbsp;&rsaquo;</a>
```
Comment (britneywwc) [inline templates/20-04/index.html:124]:
```suggestion
              <a href="/tutorials/upgrading-ubuntu-desktop#1-before-you-start">Ubuntu Desktop upgrade guide&nbsp;&rsaquo;</a>
```

File: templates/20-04/index.html [hunk 20]
```diff
@@ -0,0 +1,291 @@
...
+    quote_size="small",
+    quote_text="ESM literally saved our lives. It's allowing us to upgrade from 14.04 LTS at our own pace. It's taken the pressure off, and it also means we can tackle the Ubuntu upgrades at the same time as we roll out the new version of our platform.",
+    citation_source_name_text="Zivago Lee",
+    citation_source_title_text="Director of DevOps Engineering",
+    citation_source_organisation_text="Interana",
+    is_shallow=True
+    ) -%}
+
+    {%- if slot == 'cta' -%}
+      <a href="https://ubuntu.com/engage/interana-casestudy">Read the case study&nbsp;&rsaquo;</a>
```
Comment (britneywwc) [inline templates/20-04/index.html:258]:
```suggestion
      <a href="/engage/interana-casestudy">Read the case study&nbsp;&rsaquo;</a>
```

File: templates/20-04/index.html [hunk 21]
```diff
@@ -0,0 +1,291 @@
...
+  {% call(slot) vf_quote_wrapper(
+    quote_size="small",
+    quote_text="ESM has given us the space to plan what comes next. It's helping us get into a position where we can have a more sustainable infrastructure.",
+    citation_source_name_text="Andy Parker",
+    citation_source_title_text="Engineering Manager",
+    citation_source_organisation_text="TIM"
+    ) -%}
+
+    {%- if slot == 'cta' -%}
+      <a href="https://ubuntu.com/engage/case-study-acuris">Read the case study&nbsp;&rsaquo;</a>
```
Comment (britneywwc) [inline templates/20-04/index.html:272]:
```suggestion
      <a href="/engage/case-study-acuris">Read the case study&nbsp;&rsaquo;</a>
```

File: templates/20-04/index.html [hunk 22]
```diff
@@ -0,0 +1,291 @@
...
+    {%- endif -%}
+
+  {% endcall -%}
+
+  <hr class="is-fixed-width" />
+  <section class="p-strip is-deep">
+    <div class="u-fixed-width">
+      <h2>
+        <a href="https://ubuntu.com/security/esm#get-in-touch">Get in touch</a> if you need advice on the best path for your company.
```
Comment (britneywwc) [inline templates/20-04/index.html:281]:
```suggestion
        <a href="/security/esm#get-in-touch">Get in touch</a> if you need advice on the best path for your company.
```

**Review by britneywwc — DISMISSED** [review]
Looks great thanks, +1 with some minor suggestions

**Comment by Sophie-32** [comment]
Screen Shot 2024-12-18 at 2 30 22 PM

Will the capital H2 read the same as lowercase h2? Asking because this is the only tag thats capitalized.

**Comment by Sophie-32** [comment]
missing alt= for the images

**Comment by Sophie-32** [comment]
I pointed it out to PM, but they have two copydocs and right now the one linked here is not the correct one. This is the correct one and is the one linked in the JIra issue - https://docs.google.com/document/d/1lzxAL9Wt24m5-p5qG9y190rOg3FY5zVdKLYLlBoBcLU/edit?tab=t.0

**Comment by petesfrench** [comment]
> Screen Shot 2024-12-18 at 2 30 22 PM
> 
> Will the capital H2 read the same as lowercase h2? Asking because this is the only tag thats capitalized.

This is not capitalized for me in code or in the demo. I think it might be your browser

**Comment by petesfrench** [comment]
> missing alt= for the images

All the images have `alt=""`, is this not the correct practice to follow?

**Review by britneywwc — APPROVED** [review]

**Pull request merged**